File: p7/Daguerre/cliente.js
const net = require('net');
const readline = require('readline');

const client = new net.Socket();
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

client.on('connect', () => {
  console.log('Conectado al servidor.');
});

client.on('data', (data) => {
  console.log(data.toString());
});

client.on('close', () => {
  console.log('Desconectado del servidor.');
});

client.on('error', (err) => {
  console.error(`Error del cliente: ${err}`);
  client.destroy();
});

rl.question('Introduce tu nombre: ', (name) => {
  client.connect(4000, 'localhost', () => {
    console.log('Conectando al servidor...');
    client.write(name);
  });
});

rl.on('line', (input) => {
  client.write(input);
});

rl.on('SIGINT', () => {
  console.log('\nDesconectando del servidor...');
  client.end();
  rl.close();
});

File: p7/Daguerre/cliente.py
import socket
import threading
import sys

def receive_messages(client_socket):
    try:
        while True:
            data = client_socket.recv(1024)
            if not data:
                break
            message = data.decode('utf-8')
            print(message)
    except Exception as e:
        print(f"Error del cliente: {e}")
    finally:
        client_socket.close()
        sys.exit()

def start_client():
    client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

    try:
        client_socket.connect(('localhost', 4000))
        print("Conectando al servidor...")

        threading.Thread(target=receive_messages, args=(client_socket,)).start()

        name = input("Introduce tu nombre: ")
        client_socket.sendall(name.encode('utf-8'))

        while True:
            input_msg = input()
            client_socket.sendall(input_msg.encode('utf-8'))
    except KeyboardInterrupt:
        print("\nDesconectando del servidor...")
    finally:
        client_socket.close()

if __name__ == '__main__':
    start_client()